### DISCUSSION TITLE
Fabric-gateway doesn't provide admin-level apis.

### DISCUSSION THREAD

**Discussion by JiMa98** [hyperledger/fabric-gateway#599] [Q&A]
I am using Fabric 2.4.1 to develope my application. However, I found fabric-gateway only provides some high-level apis like invoking and querying the world state. Is there anyway except using command lines to do admin-level operations? For example, creating a new channel ,adding a peer to the channel or installing a chaincode to an endorser. Fabric-sdk-java seems to provide these apis using HFClient object, but i don't know whether it is compatible with v2.4.x. The README of Fabric-sdk-java says it is deprecated. Can I use any version of Fabric-sdk-java in my application?

**Comment by bestbeforetoday** [comment]
See [this answer](https://github.com/hyperledger/fabric-gateway/discussions/594#discussioncomment-6000514) to a similar previous discussion.

**Reply by JiMa98** [reply]
Perhaps it's a good idea to rollback to v1.4.x ? Is there any key difference between v1.4 and v2.4 ? Because learning go from scratch is a big challenge for me. Thank you.

**Reply by bestbeforetoday** [reply]
Fabric v2.5 is the current supported LTS release. The previous (v2.2) LTS release is still receiving security updates - at least through 2023. Fabric v1.4 is no longer supported and the last release was in 2021. You can see the changes introduced in the v2.x releases here:

https://hyperledger-fabric.readthedocs.io/en/latest/whatsnew.html

While no longer supported or developed, the legacy SDKs will probably mostly still work OK with Fabric v2.5. If you find the admin capability they provide is sufficient and works for your case then you could continue to use them. Using the Fabric CLI commands (or fabric-admin-sdk if using Go) for admin tasks would probably be my choice though.
